### PR TITLE
:book: Add a project roadmap doc

### LINKED ISSUE #1419: Add a project roadmap document
**Describe the solution you'd like**
As a developer, I would like to know what features are coming in the future and their rough schedule, so that I can know what to work on next.

As a user, I would like to know what features are coming in the future.

**Anything else you would like to add:**
Here is a super rough draft of a potential feature list. Please edit/comment! https://docs.google.com/document/d/1IrwQbHcD-i7pOzVKAY-uTKD2Q6hLphP7azsvSywvmcE/edit#

Once the revisions have settled down, I will turn the doc into a PR to the repo

/kind documentation
/priority important-soon

### PR DESCRIPTION
**Pull request by detiber**
**What this PR does / why we need it**:

Adds a project roadmap

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1419
/milestone v0.5.0

### CODE REVIEW

File: docs/roadmap.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+## v0.6 (v1alpha5? v1beta1?) ~ November 2020
+
+- ?
+
+## TBD
+
+- Implement MachinePools - Autoscaling groups and spot instances
+- Dual stack IPv4/IPv6 support
+- Windows nodes
+- GPU support (BitFusion, Elastic GPU, etc.) (may be provider-specific)
```
Comment (randomvariable) [inline docs/roadmap.md]:
```suggestion
- Support for GPU instances and Elastic GPU
```

File: docs/roadmap.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+- SR-IOV support (may be provider-specific, may be image-builder)
```
Comment (randomvariable) [inline docs/roadmap.md]:
```suggestion
```
Not relevant.

File: docs/roadmap.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+- [Disable creation of bastion host by default](https://github.com/kubernetes-sigs/cluster-api-provider-aws/issues/947)
+- [Ability to configure node disks](https://github.com/kubernetes-sigs/cluster-api-provider-aws/issues/1403)
+- [SSM integration (alternative to bastion host)](https://github.com/kubernetes-sigs/cluster-api-provider-aws/issues/1553)
+- [Encrypted user-data](https://github.com/kubernetes-sigs/cluster-api-provider-aws/issues/1387)
+- [Support for cross-zone load balancing](https://github.com/kubernetes-sigs/cluster-api-provider-aws/issues/1416)
+- Consume Cluster API e2e testing framework
+
+## v0.6 (v1alpha4) ~ July 2020
+
+- Private API-server load balancer
```
Comment (randomvariable) [inline docs/roadmap.md]:
```suggestion
```
I think this was done already?

File: docs/roadmap.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+- Define position on supporting multiple AWS accounts in the same manager
```
Comment (randomvariable) [inline docs/roadmap.md]:
```suggestion
- [Supporting multiple AWS accounts as Kubernetes objects](https://github.com/kubernetes-sigs/cluster-api-provider-aws/issues/1552)
```

File: docs/roadmap.md (hunk 3 above)
Comment (detiber) [reply docs/roadmap.md]: Yes, thanks for catching that!

**Comment by ncdc** [comment]
LGTM

**Comment by ncdc** [comment]
/lgtm
/hold
for lazy consensus. Expires March 5

**Comment by chuckha** [comment]
this looks great, v0.6 will be a wonderful release with the bootstrap failure detection

**Comment by detiber** [comment]
Removing hold, since lazy consensus period has ended

**Pull request merged**